### PR TITLE
GHA: Migrate semgrep action to centralised workflow

### PR DESCRIPTION
**Pull request by steelhead31**
Ref: #3371 

Migrate the infrastructure semgrep diff github action to use a centralised workflow to reduce maintenance, and reusability.

##### Checklist

- [X] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [X] VPC/QPC not applicable for this PR

### CODE REVIEW

File: .github/workflows/semgrep_diff.yml [hunk 1]
```diff
@@ -4,20 +4,6 @@ on:
...
-        uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
-        with:
-          fetch-depth: 0
-
-      # Step 2: Differential scan
-      - name: Differential scan
-        run: |
-          semgrep ci \
-            --config="p/trailofbits"
+  codefreeze:
```
Comment (gdams) [inline .github/workflows/semgrep_diff.yml:7]:
```suggestion
  semgrep-diff:
```

File: .github/workflows/semgrep_diff.yml [hunk 2]
```diff
@@ -4,20 +4,6 @@ on:
...
+    uses: adoptium/.github/.github/workflows/semgrep_diff.yml@main
+    secrets: inherit
```
Comment (gdams) [inline .github/workflows/semgrep_diff.yml:9]: are secrets needed for this workflow?

File: .github/workflows/semgrep_diff.yml (hunk 1 above)
Comment (steelhead31) [reply .github/workflows/semgrep_diff.yml]: Fixed.. :) thanks

File: .github/workflows/semgrep_diff.yml (hunk 2 above)
Comment (steelhead31) [reply .github/workflows/semgrep_diff.yml:9]: Nope, removed.

**Review by gdams — APPROVED** [review]

**Review by gdams — CHANGES_REQUESTED** [review]
Sorry, seeing errors:

```output
[Invalid workflow file: .github/workflows/semgrep_diff.yml#L8](https://github.com/adoptium/infrastructure/actions/runs/8922376330/workflow)
error parsing called workflow
".github/workflows/semgrep_diff.yml"
-> "adoptium/.github/.github/workflows/semgrep_diff.yml@main" (source branch with sha:f02a5f88fca8fc16da8947fce96a8b9a2f528830)
: workflow is not reusable as it is missing a `on.workflow_call` trigger
```

**Comment by steelhead31** [comment]
I'll take a look..

**Review by gdams — APPROVED** [review]

**Review by karianna — APPROVED** [review]

**Pull request merged**